pkg/graveler/ref/commit_iterator.go

```diff
@@ -69,20 +72,20 @@ func NewCommitIterator(ctx context.Context, db db.Database, repositoryID gravele
 	}
 }
 
-func (ci *CommitIterator) getCommitRecord(commitID graveler.CommitID) (*graveler.CommitRecord, error) {
-	var rec commitRecord
-	err := ci.db.
-		Get(ci.ctx, &rec, `SELECT id, committer, message, creation_date, parents, meta_range_id, metadata, version, generation
-			FROM graveler_commits
-			WHERE repository_id = $1 AND id = $2`,
-			ci.repositoryID, commitID)
+func (ci *KVCommitIterator) getCommitRecord(commitID graveler.CommitID) (*graveler.CommitRecord, error) {
+	commitKey := graveler.CommitPath(commitID)
+	c := graveler.CommitData{}
+	_, err := ci.kvStore.GetMsg(ci.ctx, graveler.CommitPartition(ci.repositoryID), []byte(commitKey), &c)
 	if err != nil {
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrCommitNotFound
```

Comment:
I don't think this is a proper error since the user wasn't asking for a specific commit while trying to get the log of the commits.

Reply:
changed to ErrNotFound